crates/solidity/inputs/language/src/lib.rs

```diff
@@ -1,24 +1,20 @@
-use std::path::PathBuf;
-
 use anyhow::Result;
 use codegen_schema::types::{LanguageDefinition, LanguageDefinitionRef};
 
 pub trait SolidityLanguageExtensions {
     fn load_solidity() -> Result<LanguageDefinitionRef>;
 }
 
+// Set by the build script.
+static LANGUAGE_DEFINITION_BIN: &'static str = env!("SLANG_SOLIDITY_LANGUAGE_DEFINITION_BIN");
```

Comment:
> These seem to be redundant: any crate using the `solidity_language::SolidityLanguageExtensions` will by rebuilt if the `solidity_language` crate changes but it was that package's build script that defined the variable that was explicitly depended upon by the callers, so no need to manually instruct Cargo here.

You are correct that dependencies will get rebuilt if `solidity_language` source is changed, but not the output. Without this flag, if you delete the binary file (or simply do `cargo clean`) your build will be broken. `solidity_language` won't rebuild, and its dependencies will crash trying to read a non-existing file.
To safeguard against this, we should always mark both inputs and outputs of any build step.

Reply:
Manually deleting the build can break the build, you're right, but cargo clean will always create the relevant file as part of the regular build process, this doesn't break this (and I tested this).

To correctly re-compile the language definition I added the `rerun-if-changed` directive to the build script that creates the file - downstream build scripts do not need to emit that, as they will always be re-run if the dependency will be dirty (in this case the solidity_language build script and so also the library target) in 514d88919b3f6e58a9d9d6a22ad32eb0f0e60aa0.

Reply:
I see. Thanks for fixing it!